### PR TITLE
Optimize Bondi-Michel and Kerr-Schild Solutions

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

- Reduces the number of allocations in the Bondi-Michel accretion solution
- Reduces the number of allocations in Kerr-Schild
- Clean up Local LF flux.

**Note:** Only last 3 commits are new.

Depends on:
- [x] #1137 
- [x] #1160 

### Types of changes:

- [ ] Bugfix
- [x] New feature

### Component:

- [x] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by wthrowe** [comment]
The changes here look good to me.  Just need to wait for the dependencies to get merged.

**Comment by nilsdeppe** [comment]
Rebased on top of latest #1160

**Comment by nilsdeppe** [comment]
Rebased on develop. This is ready to review :)

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.cpp
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.cpp:165]: Update the comment above to reflect the new variable name.

I suggest changing line (new) 162 to say "Below we use `spin_a` = the" and (new) line 164 to remove "So compute spin=a here".

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.cpp
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.cpp]: What is this free-floating comment describing? What needs the items listed below? Please clarify what this is trying to communicate.

File: src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.cpp
Comment (fmahebert) [inline src/PointwiseFunctions/AnalyticSolutions/GeneralRelativity/KerrSchild.cpp]: I appreciate this comment, but it bothers me to justify the DataVector specialization with ray tracing as an argument, because ray tracing is much more likely to use the double specialization. Perhaps remove the ray-tracing argument altogether? Or move this comment somewhere that it applies to both the double and DataVector cases.

**Review by fmahebert — CHANGES_REQUESTED** [review]

**Comment by nilsdeppe** [comment]
Changed comments

**Review by fmahebert — APPROVED** [review]

**Comment by nilsdeppe** [comment]
@kidder @wthrowe This should be ready (@wthrowe already looked at it before the dependencies were merged :) )

**Review by kidder — APPROVED** [review]

**Review by wthrowe — APPROVED** [review]

**Pull request merged**
